File: apps/api/BHMHockey.Api/Models/Entities/TournamentAdmin.cs
namespace BHMHockey.Api.Models.Entities;

public class TournamentAdmin
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid TournamentId { get; set; }
    public Tournament Tournament { get; set; } = null!;

    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    // Role: Owner, Admin, Scorekeeper
    public string Role { get; set; } = "Admin";

    public Guid? AddedByUserId { get; set; }
    public User? AddedByUser { get; set; }

    public DateTime AddedAt { get; set; } = DateTime.UtcNow;
    public DateTime? RemovedAt { get; set; }
}

File: packages/shared/src/types/tournamentAdmin.ts
// ============================================
// Tournament Admin Types (TRN-028)
// ============================================

// Tournament admin roles
export type TournamentAdminRole = 'Owner' | 'Admin' | 'Scorekeeper';

// Tournament admin DTO - response for listing admins
export interface TournamentAdminDto {
  id: string;
  tournamentId: string;
  userId: string;
  userFirstName: string;
  userLastName: string;
  userEmail: string;
  role: TournamentAdminRole;
  addedAt: string;
  addedByUserId: string | null;
  addedByName: string | null;
}

// Request to add a tournament admin
export interface AddTournamentAdminRequest {
  userId: string;
  role: TournamentAdminRole;  // Cannot be 'Owner' - ownership is only transferable
}

// Request to update tournament admin role
export interface UpdateTournamentAdminRoleRequest {
  role: TournamentAdminRole;  // Cannot be 'Owner' - ownership is only transferable
}

// Request to transfer tournament ownership
export interface TransferOwnershipRequest {
  newOwnerUserId: string;
}
